doc/source/data/contributing/how-to-contribute-an-improvement.md

```diff
@@ -0,0 +1,67 @@
+# How to contribute an improvement
+
+If you want your changes to be reviewed and merged quickly, following a few key 
+practices makes a big difference. Clear, focused, and well-structured contributions help 
+reviewers understand your intent and ensure your improvements land smoothly.
+
+## Find something to work on
+
+Start by solving a problem you encounter, like fixing a bug or adding a missing feature. 
+If you're unsure where to start:
+* Browse the issue tracker for problems you understand.
+* Look for labels like "good first issue" for approachable tasks.
+* Join the Ray Slack and message @richardliaw.
+
+## Get early feedback
+
+If you’re adding a new public API or making a substantial refactor, 
+**share your plan early**. Discussing changes before you invest a lot of work can save 
+time and align your work with the project’s direction.
+
+You can open a draft PR, discuss on an Issue, or post in Slack for early feedback. It 
+won’t affect acceptance and often improves the final design.
+
+## Write a clear pull request description
+
+Explain **why the change exists and what it achieves**. Clear descriptions reduce 
+back-and-forth and speed up reviews.
+
+## Keep pull requests small
```

Comment:
For both of these, suggest to link to a positive example PR to make it concrete (but don't pick a negative one to shame the author)